src/Mutant/Generator/MutationsGenerator.php

```diff
@@ -126,17 +128,33 @@ private function getMutationsFromFile(SplFileInfo $file, bool $onlyCovered, arra
         \assert(\is_string($filePath));
 
         $mutationsCollectorVisitor = new MutationsCollectorVisitor(
-            $mutators,
+            $this->mutators,
             $filePath,
             $initialStatements,
             $this->codeCoverageData,
             $onlyCovered
         );
 
-        $traverser->addVisitor(new ParentConnectorVisitor());
-        $traverser->addVisitor(new FullyQualifiedClassNameVisitor());
-        $traverser->addVisitor(new ReflectionVisitor());
-        $traverser->addVisitor($mutationsCollectorVisitor);
+        $orderedVisitors = [
+            40 => new ParentConnectorVisitor(),
+            30 => new FullyQualifiedClassNameVisitor(),
+            20 => new ReflectionVisitor(),
+            10 => $mutationsCollectorVisitor,
+        ];
+
+        $visitorsQueue = new \SplPriorityQueue();
```

Comment:
I think we should move this logic out of this class. The ordering that is.

I think we can extend the NodeTraverser, and maybe opt to open an issue or PR on php-parser to get that functionality in there.

Reply:
> I think we should move this logic out of this class.

could you please elaborate on this? move to to which class? how would it look like? what are the benefits? where to store priorities?

> I think we can extend the NodeTraverser

again - what for? I'm not a big fan of extending (read - overriding) lib's classes, so I would like to understand the purpose here.

Reply:
> My idea would be to move this to a new class.

Done https://github.com/infection/infection/pull/491/commits/623906c13be7b5e85fb38f58d3b3e417c415a614

> That is because it is not the responsibility of this class to determine the order in which node visitors should be used.

The order of visitors was determined before this PR (in master) in this class, by ordering `->addVistor()` method calls

```php
$traverser->addVisitor(new ParentConnectorVisitor());
$traverser->addVisitor(new FullyQualifiedClassNameVisitor());
$traverser->addVisitor(new ReflectionVisitor());
$traverser->addVisitor($mutationsCollectorVisitor);
```

and I think this is a correct place to do that, because it is how PHP-Parser NodeTraverser's API works - we create an object and adds ordered visitors to influence to the AST.

If you mean we should move `->addVisitor()` method calls to one more class - I would say this would be an overkill